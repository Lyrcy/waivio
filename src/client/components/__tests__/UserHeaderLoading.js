import React from 'react';
import { act } from 'react-dom/test-utils';
import { shallow, mount } from 'enzyme';
import UserHeaderLoading from '../UserHeaderLoading';

describe('UserHeaderLoading component', () => {
  let wrapper;
  let UserHeaderLoadingAvatar;
  let AntCardLoadingBlock;

  beforeEach(() => {
    wrapper = mount(<UserHeaderLoading />);

    act(() => {
      wrapper.update();
    });

    UserHeaderLoadingAvatar = wrapper.find('.UserHeaderLoading__avatar');
    AntCardLoadingBlock = wrapper.find('.ant-card-loading-block');
  });

  test('Should not to be undefined', () => {
    expect(wrapper).not.toBeUndefined();
  });

  test('should render without exploding', () => {
    wrapper = shallow(<UserHeaderLoading />);
    act(() => {
      wrapper.update();
    });
    expect(wrapper).toMatchSnapshot();
  });

  test('Should not to be undefined components', () => {
    expect(UserHeaderLoadingAvatar).not.toBeUndefined();
    expect(AntCardLoadingBlock).not.toBeUndefined();
  });
});
