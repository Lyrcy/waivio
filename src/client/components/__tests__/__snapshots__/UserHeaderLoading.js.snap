// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserHeaderLoading component should render without exploding 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserHeaderLoading />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="UserHeaderLoading__container"
      >
        <div
          className="ant-card-loading-block UserHeaderLoading__avatar"
          style={
            Object {
              "height": "100px",
              "width": "100px",
            }
          }
        />
        <div
          className="UserHeaderLoading__user"
        >
          <div
            className="ant-card-loading-block"
            style={
              Object {
                "height": "32px",
                "marginBottom": 0,
                "marginLeft": 0,
                "width": "160px",
              }
            }
          />
          <div
            className="ant-card-loading-block"
            style={
              Object {
                "height": "20px",
                "marginLeft": 0,
                "width": "120px",
              }
            }
          />
          <div
            className="UserHeaderLoading__user"
          >
            <div
              className="ant-card-loading-block"
              style={
                Object {
                  "height": "20px",
                  "marginLeft": 0,
                  "width": "100px",
                }
              }
            />
          </div>
        </div>
      </div>,
      "className": "UserHeaderLoading",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="ant-card-loading-block UserHeaderLoading__avatar"
            style={
              Object {
                "height": "100px",
                "width": "100px",
              }
            }
          />,
          <div
            className="UserHeaderLoading__user"
          >
            <div
              className="ant-card-loading-block"
              style={
                Object {
                  "height": "32px",
                  "marginBottom": 0,
                  "marginLeft": 0,
                  "width": "160px",
                }
              }
            />
            <div
              className="ant-card-loading-block"
              style={
                Object {
                  "height": "20px",
                  "marginLeft": 0,
                  "width": "120px",
                }
              }
            />
            <div
              className="UserHeaderLoading__user"
            >
              <div
                className="ant-card-loading-block"
                style={
                  Object {
                    "height": "20px",
                    "marginLeft": 0,
                    "width": "100px",
                  }
                }
              />
            </div>
          </div>,
        ],
        "className": "UserHeaderLoading__container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "ant-card-loading-block UserHeaderLoading__avatar",
            "style": Object {
              "height": "100px",
              "width": "100px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="ant-card-loading-block"
                style={
                  Object {
                    "height": "32px",
                    "marginBottom": 0,
                    "marginLeft": 0,
                    "width": "160px",
                  }
                }
              />,
              <div
                className="ant-card-loading-block"
                style={
                  Object {
                    "height": "20px",
                    "marginLeft": 0,
                    "width": "120px",
                  }
                }
              />,
              <div
                className="UserHeaderLoading__user"
              >
                <div
                  className="ant-card-loading-block"
                  style={
                    Object {
                      "height": "20px",
                      "marginLeft": 0,
                      "width": "100px",
                    }
                  }
                />
              </div>,
            ],
            "className": "UserHeaderLoading__user",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "ant-card-loading-block",
                "style": Object {
                  "height": "32px",
                  "marginBottom": 0,
                  "marginLeft": 0,
                  "width": "160px",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "ant-card-loading-block",
                "style": Object {
                  "height": "20px",
                  "marginLeft": 0,
                  "width": "120px",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="ant-card-loading-block"
                  style={
                    Object {
                      "height": "20px",
                      "marginLeft": 0,
                      "width": "100px",
                    }
                  }
                />,
                "className": "UserHeaderLoading__user",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "ant-card-loading-block",
                  "style": Object {
                    "height": "20px",
                    "marginLeft": 0,
                    "width": "100px",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="UserHeaderLoading__container"
        >
          <div
            className="ant-card-loading-block UserHeaderLoading__avatar"
            style={
              Object {
                "height": "100px",
                "width": "100px",
              }
            }
          />
          <div
            className="UserHeaderLoading__user"
          >
            <div
              className="ant-card-loading-block"
              style={
                Object {
                  "height": "32px",
                  "marginBottom": 0,
                  "marginLeft": 0,
                  "width": "160px",
                }
              }
            />
            <div
              className="ant-card-loading-block"
              style={
                Object {
                  "height": "20px",
                  "marginLeft": 0,
                  "width": "120px",
                }
              }
            />
            <div
              className="UserHeaderLoading__user"
            >
              <div
                className="ant-card-loading-block"
                style={
                  Object {
                    "height": "20px",
                    "marginLeft": 0,
                    "width": "100px",
                  }
                }
              />
            </div>
          </div>
        </div>,
        "className": "UserHeaderLoading",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="ant-card-loading-block UserHeaderLoading__avatar"
              style={
                Object {
                  "height": "100px",
                  "width": "100px",
                }
              }
            />,
            <div
              className="UserHeaderLoading__user"
            >
              <div
                className="ant-card-loading-block"
                style={
                  Object {
                    "height": "32px",
                    "marginBottom": 0,
                    "marginLeft": 0,
                    "width": "160px",
                  }
                }
              />
              <div
                className="ant-card-loading-block"
                style={
                  Object {
                    "height": "20px",
                    "marginLeft": 0,
                    "width": "120px",
                  }
                }
              />
              <div
                className="UserHeaderLoading__user"
              >
                <div
                  className="ant-card-loading-block"
                  style={
                    Object {
                      "height": "20px",
                      "marginLeft": 0,
                      "width": "100px",
                    }
                  }
                />
              </div>
            </div>,
          ],
          "className": "UserHeaderLoading__container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "ant-card-loading-block UserHeaderLoading__avatar",
              "style": Object {
                "height": "100px",
                "width": "100px",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="ant-card-loading-block"
                  style={
                    Object {
                      "height": "32px",
                      "marginBottom": 0,
                      "marginLeft": 0,
                      "width": "160px",
                    }
                  }
                />,
                <div
                  className="ant-card-loading-block"
                  style={
                    Object {
                      "height": "20px",
                      "marginLeft": 0,
                      "width": "120px",
                    }
                  }
                />,
                <div
                  className="UserHeaderLoading__user"
                >
                  <div
                    className="ant-card-loading-block"
                    style={
                      Object {
                        "height": "20px",
                        "marginLeft": 0,
                        "width": "100px",
                      }
                    }
                  />
                </div>,
              ],
              "className": "UserHeaderLoading__user",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "ant-card-loading-block",
                  "style": Object {
                    "height": "32px",
                    "marginBottom": 0,
                    "marginLeft": 0,
                    "width": "160px",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "ant-card-loading-block",
                  "style": Object {
                    "height": "20px",
                    "marginLeft": 0,
                    "width": "120px",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="ant-card-loading-block"
                    style={
                      Object {
                        "height": "20px",
                        "marginLeft": 0,
                        "width": "100px",
                      }
                    }
                  />,
                  "className": "UserHeaderLoading__user",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "ant-card-loading-block",
                    "style": Object {
                      "height": "20px",
                      "marginLeft": 0,
                      "width": "100px",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
